bigtable/snippets/src/main/java/com/example/bigtable/Filters.java

```diff
@@ -0,0 +1,360 @@
+  }
+  // [END bigtable_filters_limit_cells_per_row_offset]
+
+  // [START bigtable_filters_limit_col_family_regex]
+  public static void filterLimitColFamilyRegex(
+      String projectId, String instanceId, String tableId) {
+    // String projectId = "my-project-id";
+    // String instanceId = "my-instance-id";
+    // String tableId = "mobile-time-series";
+
+    Filter filter = FILTERS.family().regex("stats_.*$");
+    readFilter(projectId, instanceId, tableId, filter);
+  }
+  // [END bigtable_filters_limit_col_family_regex]
+
+  // [START bigtable_filters_limit_col_qualifier_regex]
+  public static void filterLimitColQualifierRegex(
+      String projectId, String instanceId, String tableId) {
+    // String projectId = "my-project-id";
+    // String instanceId = "my-instance-id";
+    // String tableId = "mobile-time-series";
+
+    Filter filter = FILTERS.qualifier().regex("connected_.*$");
+    readFilter(projectId, instanceId, tableId, filter);
+  }
+  // [END bigtable_filters_limit_col_qualifier_regex]
+
+  // [START bigtable_filters_limit_col_range]
+  public static void filterLimitColRange(String projectId, String instanceId, String tableId) {
+    // String projectId = "my-project-id";
+    // String instanceId = "my-instance-id";
+    // String tableId = "mobile-time-series";
+
+    Filter filter =
+        FILTERS
+            .qualifier()
+            .rangeWithinFamily("cell_plan")
+            .startClosed("data_plan_01gb")
+            .endOpen("data_plan_10gb");
+    readFilter(projectId, instanceId, tableId, filter);
+  }
+  // [END bigtable_filters_limit_col_range]
+
+  // [START bigtable_filters_limit_value_range]
+  public static void filterLimitValueRange(String projectId, String instanceId, String tableId) {
+    // String projectId = "my-project-id";
+    // String instanceId = "my-instance-id";
+    // String tableId = "mobile-time-series";
+
+    Filter filter = FILTERS.value().range().startClosed("PQ2A.190405").endClosed("PQ2A.190406");
+    readFilter(projectId, instanceId, tableId, filter);
+  }
+  // [END bigtable_filters_limit_value_range]
+
+  // [START bigtable_filters_limit_value_regex]
+  public static void filterLimitValueRegex(String projectId, String instanceId, String tableId) {
+    // String projectId = "my-project-id";
+    // String instanceId = "my-instance-id";
+    // String tableId = "mobile-time-series";
+
+    Filter filter = FILTERS.value().regex("PQ2A.*$");
+    readFilter(projectId, instanceId, tableId, filter);
+  }
+  // [END bigtable_filters_limit_value_regex]
+
+  // [START bigtable_filters_limit_timestamp_range]
+  public static void filterLimitTimestampRange(
+      String projectId, String instanceId, String tableId) {
+    // String projectId = "my-project-id";
+    // String instanceId = "my-instance-id";
+    // String tableId = "mobile-time-series";
+    long timestamp = System.currentTimeMillis() * 1000;
+    long hour = 60L * 60 * 1000 * 1000;
```

Comment:
You should use the [`java.time`](https://github.com/GoogleCloudPlatform/java-docs-samples/blob/master/SAMPLE_FORMAT.md#function-structure) package to deal with time:
```java
// Get a time representing one hour ago
Instant timestamp = Instant.now().minus(1, ChronoUnit.HOURS);

Filter filter = FILTERS.timestamp().range().startClosed(0L).endOpen(timestamp.toEpochMilli());
```